core/templates/pages/exploration-editor-page/translation-tab/services/translation-topic.service.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview A service that maintains a record of which topic
+ * in the translation tab is currently active.
+ */
+
+import { EventEmitter, Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+
+import { ContributionOpportunitiesService } from
+  // eslint-disable-next-line max-len
+  'pages/contributor-dashboard-page/services/contribution-opportunities.service';
+import { LoggerService } from 'services/contextual/logger.service';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TranslationTopicService {
+  private activeTopicName: string = null;
+  private _activeTopicChangedEventEmitter = new EventEmitter<void>();
+
+  constructor(
+    private ContributionOpportunitiesService: ContributionOpportunitiesService,
+    private loggerService: LoggerService) {}
+
+  getActiveTopicName(): string {
+    return this.activeTopicName;
+  }
+
+  setActiveTopicName(newActiveTopicName: string): void {
+    this.ContributionOpportunitiesService.getAllTopicNamesAsync()
+      .then((data) => {
+        if (newActiveTopicName !== 'All' &&
+            data.indexOf(newActiveTopicName) < 0) {
+          this.loggerService.error(
+            'Invalid active topic name: ' + newActiveTopicName);
+          return;
+        }
+        this.activeTopicName = newActiveTopicName;
+        this._activeTopicChangedEventEmitter.emit();
+      });
+  }
+
+  get onActiveTopicChanged(): EventEmitter<void> {
+    return this._activeTopicChangedEventEmitter;
+  }
```

Comment:
Do you think this would be better named `getActiveTopicChangedEventEmitter` for clarity?

Reply:
This emitter is triggered when the active topic is changed, so I personally think `_activeTopicChangedEventEmitter` would be more accurate.